### PR TITLE
Vet clinic database: database performance audit

### PR DESCRIPTION
**Pull request by iAmWaris97**
In this stage of the vet_clinic database project we have done the following:

- Used the schema and data from the previous projects.
- Add email column into owners table.
- Inserted huge amounts of data into visits and owners tables to check their performance.
- Checked the performance by running the queries with the "explain analyze".

**Followings are the queries before and after creating INDEX on tables columns** 

 explain analyze SELECT COUNT(*) FROM visits where animals_id = 4; ==> Before  CREATE INDEX

explain analyze SELECT COUNT(*) FROM visits where animals_id = 4; ==> After CREATE INDEX

a1

explain analyze SELECT * FROM visits where vet_id = 2; Before  CREATE INDEX

explain analyze SELECT * FROM visits where vet_id = 2; After CREATE INDEX

a2

explain analyze SELECT * FROM owners where email = 'owner_18327@mail.com'; Before  CREATE INDEX

explain analyze SELECT * FROM owners where email = 'owner_18327@mail.com'; After CREATE INDEX

a3

### CODE REVIEW

File: schema.sql [hunk 1]
```diff
@@ -37,4 +37,30 @@ ALTER TABLE animals
...
+CREATE TABLE visits (
+    id INT primary key generated always as identity,
+    vets_id INT NULL REFERENCES vets(id),
+	animals_id INT NULL REFERENCES animals(id),
+	visit_date DATE NULL
+);
+
+/* Block-2 */
+/*Day-1*/
+ALTER TABLE owners ADD COLUMN email VARCHAR(120);
```
Comment (SajjadAhmad14) [inline schema.sql:64]: For `SELECT COUNT(*) FROM visits where animals_id = 4;` and `SELECT * FROM visits where vet_id = 2` queries:

The execution time should be significantly lower than 1 sec (1000 ms). Kindly add the indexes again and re-run the db to see if the performance of queries increases.

File: schema.sql [hunk 2]
```diff
@@ -37,4 +37,30 @@ ALTER TABLE animals
...
 
+-- Create index to increase performance speed
```
Comment (SajjadAhmad14) [inline schema.sql:65]: Kindly add indexes to `visits` and `owners` here in the schema file.
 ```sql
   CREATE INDEX ON visits (vet_id);
   CREATE INDEX ON owners (email);
```

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi team, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
Git flow is followed. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: schema.sql (hunk 1 above)
Comment (iAmWaris97) [reply schema.sql:64]: Hi,

Thank you for the kind review,

During inserting data into tables, we have inserted a huge amount of data, and as you can see CREATE INDEX decreases the time all the way from 6.5 seconds to 1.8 seconds.
And CREATE INDEX cannot be done twice on the same table and column.

File: schema.sql (hunk 2 above)
Comment (iAmWaris97) [reply schema.sql:65]: Sorry for the mistake which happened here!

I was added but during the conflict resolving was removed by accident.
Added again.

**Review by Reem-lab — APPROVED** [review]
# Hi @fauziyahzeeyah,👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉🔥🔥💯💯



## Highlights ✨✨
📌 Descriptive PR title and summary ✅
📌 Great working on the functionality ✅


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**